### PR TITLE
[Hotfix] Fix Issue #506 and Issue #505

### PR DESCRIPTION
**Pull request by kevin-zhonghao**
https://github.com/futurewei-cloud/alcor/issues/505
https://github.com/futurewei-cloud/alcor/issues/506

### CODE REVIEW

File: services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java [hunk 1]
```diff
@@ -523,8 +523,10 @@ public ResponseId deleteSubnetState(@PathVariable String projectId, @PathVariabl
...
                 rangeId = ipV6RangeId;
             }
-//            Boolean checkIfAnyPortInSubnet = this.subnetService.checkIfAnyPortInSubnet(rangeId);
-//            if (checkIfAnyPortInSubnet) {
+
+            // TODO: check if there is any gateway / non-gateway port for the subnet, waiting for PM new API
+//            Boolean checkIfAnyNoneGatewayPortInSubnet = this.subnetService.checkIfAnyPortInSubnet(rangeId);
+//            if (checkIfAnyNoneGatewayPortInSubnet) {
 //                throw new HavePortInSubnet();
 //            }
```
Comment (cj-chung) [inline services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java:531]: @kevin-zhonghao Please check PM's new API for port count in a subnet, it should be `{"/project/{project_id}/subnet-port-number/{subnet_id}`. This api will only return non-gateway port count. If it's not zero, the subnet cannot be deleted. If it returns zero, we need to ask PM to delete the gateway port first before delete the subnet.

**Review by cj-chung — CHANGES_REQUESTED** [review]

File: services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterToPMServiceImpl.java [hunk 2]
```diff
@@ -81,4 +81,10 @@ public void updatePort(String projectid, String portId, PortEntity portEntity) {
         HttpEntity<PortWebJson> request = new HttpEntity<>(new PortWebJson(portEntity));
         restTemplate.put(portManagerServiceUrl, request, PortWebJson.class);
     }
+
+    @Override
+    public void updateL3Neighbors(String projectid, List<String> gatewayPorts) {
+        String portManagerServiceUrl = portUrl + "/project/" + projectid + "/update-l3-neighbors";
+        //restTemplate.put(portManagerServiceUrl, request, PortWebJson.class);
+    }
```
Comment (cj-chung) [inline services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterToPMServiceImpl.java:89]: @kevin-zhonghao please prepare a `RouterUpdateInfo` object here for request body. @chenpiaoping's code for this API requires `RouterUpdateInfo` object.

**Review by cj-chung — CHANGES_REQUESTED** [review]

File: services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java (hunk 1 above)
Comment (kevin-zhonghao) [reply services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java:531]: Sure

File: services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterToPMServiceImpl.java (hunk 2 above)
Comment (kevin-zhonghao) [reply services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterToPMServiceImpl.java]: Added it

**Review by cj-chung — APPROVED** [review]
LGTM

**Pull request merged**